src/details/ArborX_DetailsTreeNodeLabelling.hpp

```diff
@@ -22,64 +22,69 @@ namespace Experimental
 {
 
 template <class ExecutionSpace, class BVH, class LabelsIn, class LabelsOut>
-void initBVHLabels(ExecutionSpace const &exec_space, BVH const &bvh,
-                   LabelsIn const &in, LabelsOut &out)
+void initLabels(ExecutionSpace const &exec_space, BVH const &bvh,
+                LabelsIn const &in, LabelsOut &out)
 {
   auto const n = bvh.size();
 
+  ARBORX_ASSERT(n >= 2);
   ARBORX_ASSERT(in.size() == n);
   ARBORX_ASSERT(out.size() == 2 * n - 1);
 
   Kokkos::parallel_for(
-      "ArborX::Experimental::init_labels",
+      "ArborX::initialize_leaf_node_labels",
```

Comment:
Why? Are you thinking about the future when this is the first class citizen?

Reply:
It is in `Details::` so users are not allowed to call it directly.  We can mark the enclosing region as experimental if we think it is necessary.